client-ui/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -13,200 +13,132 @@
-            this.queryParams = { ...this.queryParams, timestampStart: state.minDate, timestampEnd: state.maxDate }
-            this.store.dispatch(new RouterAction.ReplaceUrlState(this.queryParams))
-            this.store.dispatch(new ExecutionEventsActions.Get({ ...this.queryParams, asAtTime: this.asAtTime }))
+      route.queryParams
+        .subscribe(_.bind(this.urlStatePersister.restoreState, this)),
+
+      this.store.select('dashboard')
+        .subscribe((dashboardState) => {
+          console.log("UPDATED STATE", dashboardState)
+          this.dashboardState = dashboardState
+          this.sort = {
+            prop: dashboardState.sort.field,
+            dir: dashboardState.sort.direction
           }
-        })
-    )
+          this.loadExecutionEvents(dashboardState)
+          this.loadingIndicator.activate()
+          this.urlStatePersister.persistState(dashboardState)
+        }),
 
-    this.subscriptions.push(
       this.store.select('executionEvents')
-        .pipe(
-          filter(state => state != null)
-        ).subscribe(executionEvents => {
-          this.rows = executionEvents.elements
-          this.queryParams = { ...this.queryParams, offset: executionEvents.offset }
-          this.totalCount = executionEvents.totalCount
+        .pipe(filter(_.identity))
+        .subscribe(executionEvents => {
+          console.log("received data", executionEvents)
+          this.executionEvents = executionEvents
+          this.loadingIndicator.deactivate()
         })
     )
   }
 
-  ngAfterViewInit(): void {
-
-    this.subscriptions.push(
-      this.table.page.pipe(
-        tap(_ => this.loading = true),
-      ).subscribe(
-        page => {
-          this.queryParams = { ...this.queryParams, offset: page.offset }
-          this.applyFilters()
-        }
-      )
-    )
-
-    this.subscriptions.push(
-      this.table.sort.pipe(
-        tap(_ => this.loading = true),
-        map(event => event.sorts[0]),
-      ).subscribe(
-        sort => {
-          this.queryParams = { ...this.queryParams, offset: 0, sortName: sort.prop, sortDirection: sort.dir }
-          this.applyFilters()
-        }
-      )
-    )
-
-    this.subscriptions.push(
-      fromEvent<any>(this.searchInput.nativeElement, 'keyup')
-        .pipe(
-          tap(_ => this.loading = true),
-          map(event => event.target.value),
-          debounceTime(400)
-        ).subscribe(
-          searchTerm => {
-            this.queryParams = { ...this.queryParams, offset: 0, searchTerm: searchTerm }
-            this.applyFilters()
-          }
-        )
-    )
-
-  }
-
-  private applyFilters(): void {
-    if (this.useDateRange) {
-      this.store.dispatch(new RouterAction.ReplaceUrlState(this.queryParams))
-    }
-    this.store.dispatch(new ExecutionEventsActions.Get({ ...this.queryParams, asAtTime: this.asAtTime }))
-    this.loading = false
+  private loadExecutionEvents(dashboardState): void {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected parameter: 'dashboardState' to have a typedef](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)